financial-services-accelerator/components/org.wso2.financial.services.accelerator.event.notifications.service/src/main/java/org/wso2/financial/services/accelerator/event/notifications/service/util/EventNotificationServiceUtil.java

```diff
@@ -0,0 +1,156 @@
+//                (RealtimeEventNotificationRequestGenerator) FinancialServicesUtils
+//                        .getClassInstanceFromFQN(FinancialServicesConfigParser.getInstance().
+//                        getRealtimeEventNotificationRequestGenerator());
+//        return realtimeEventNotificationRequestGenerator;
+//    }
+
+    /**
+     * Method to modify event notification payload with custom eventValues.
+     *
+     * @param jsonNode Json Node to convert
+     * @return String eventNotificationPayload
+     */
+    public static String getCustomNotificationPayload(JsonNode jsonNode) {
+
+        return jsonNode.toString();
+    }
+
+    /**
+     * Method to get event JSON from eventInformation payload string.
+     * @param eventInformation String event Information
+     * @return JSONObject converted event json
+     * @throws JSONException  Exception when parsing event information
+     */
+    public static JSONObject getEventJSONFromString(String eventInformation) throws JSONException {
+
+        return new JSONObject(eventInformation);
+    }
+
+    /**
+     * Validate if the client ID is existing.
+     * @param clientId  client ID of the TPP
+     * @throws FSEventNotificationException  Exception when validating client ID
+     */
+    @Generated(message = "Excluded since this needs OAuth2Util service provider")
+    public static void validateClientId(String clientId) throws FSEventNotificationException {
+
+        if (StringUtils.isNotEmpty(clientId)) {
+            Optional<ServiceProvider> serviceProvider;
+            try {
+                serviceProvider = Optional.ofNullable(OAuth2Util.getServiceProvider(clientId));
+                if (!serviceProvider.isPresent()) {
+                    log.error(EventNotificationConstants.INVALID_CLIENT_ID);
+                    throw new FSEventNotificationException(EventNotificationConstants.INVALID_CLIENT_ID);
+                }
+            } catch (IdentityOAuth2Exception e) {
+                log.error(EventNotificationConstants.INVALID_CLIENT_ID, e);
+                throw new FSEventNotificationException(EventNotificationConstants.INVALID_CLIENT_ID);
+            }
+        }
+    }
+
+    @Generated(message = "Creating a single instance for ConsentCoreService")
+    public static synchronized ConsentCoreServiceImpl getConsentCoreServiceImpl() {
+        if (consentCoreService == null) {
+            synchronized (ConsentCoreServiceImpl.class) {
+                if (consentCoreService == null) {
+                    consentCoreService = new ConsentCoreServiceImpl();
+                }
+            }
+        }
+        return consentCoreService;
+    }
+
+    /**
+     * Get the callback URL of the TPP from the Subscription Object.
+     *
+     * @param clientID client ID of the TPP
+     * @return callback URL of the TPP
+     */
+    public static String getCallbackURL(String clientID) {
+
+        return "http://localhost:8080/sample-tpp-server";
```

Comment:
Why are we returning a hardcoded url here?

Reply:
This is fixed from https://github.com/wso2/financial-services-accelerator/pull/193